compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -986,30 +1015,56 @@ useSingleMethodDict env (Cast ghcExpr _) f = do
 useSingleMethodDict env x _ =
     unhandled "useSingleMethodDict: not a single method type class dictionary" x
 
-convertImplements :: Env -> ModuleContents -> LF.TypeConName -> ConvertM (NM.NameMap TemplateImplements)
-convertImplements env mc tpl = NM.fromList <$>
-  mapM convertInterface (MS.findWithDefault [] tpl (mcImplements mc))
+convertTemplateImpls :: Env -> ModuleContents -> Qualified LF.TypeConName -> ConvertM (NM.NameMap TemplateImplements)
+convertTemplateImpls env mc qTplTypeCon = NM.fromList <$>
+  mapM convertTemplateImpl (MS.findWithDefault [] (qualObject qTplTypeCon) (mcImplements mc))
   where
-    convertInterface :: (Maybe LF.SourceLoc, GHC.TyCon) -> ConvertM TemplateImplements
-    convertInterface (originLoc, iface) = withRange originLoc $ do
+    convertTemplateImpl :: (Maybe LF.SourceLoc, GHC.TyCon) -> ConvertM TemplateImplements
+    convertTemplateImpl (originLoc, iface) = withRange originLoc $ do
       let handleIsNotInterface tyCon =
             "cannot implement '" ++ prettyPrint tyCon ++ "' because it is not an interface"
-      con <- convertInterfaceTyCon env handleIsNotInterface iface
-      let mod = nameModule (getName iface)
+      qIfaceTypeCon <- convertInterfaceTyCon env handleIsNotInterface iface
 
       -- TODO: Stub view, will extract later, https://github.com/digital-asset/daml/pull/14439
-      let view = ETmLam (ExprVarName "this", TCon con) (EBuiltin BEUnit)
+      let view = ETmLam (ExprVarName "this", TCon qIfaceTypeCon) (EBuiltin BEUnit)
 
-      methods <- convertMethods $ MS.findWithDefault []
-        (mod, qualObject con, tpl)
+      methods <- convertImplMethods env TemplateImplementsMethod $ MS.findWithDefault []
+        (qTplTypeCon, qIfaceTypeCon)
         (mcInterfaceMethodInstances mc)
 
-      pure (TemplateImplements con methods view)
+      pure (TemplateImplements qIfaceTypeCon methods view)
 
-    convertMethods ms = fmap NM.fromList . sequence $
-      [ TemplateImplementsMethod (MethodName k) . (`ETmApp` EVar this) <$> convertExpr env v
-      | (k, v) <- ms
-      ]
+convertInterfaceCoImpls :: Env -> ModuleContents -> Qualified LF.TypeConName -> ConvertM (NM.NameMap InterfaceCoImplements)
```

Comment:
this is almost identical to `convertTemplateImpls` but it's a pain to refactor them into a generic conversion function, so I left the redundancy